src/views/v3/Bridge/AmountInput/index.tsx

```diff
@@ -0,0 +1,192 @@
+    );
+  },
+);
+
+type Props = {
+  value: string;
+  debauncedValue: string;
+  supportedSourceTokens: Array<Token>;
+  tokenBalance: sdkAmount.Amount | null;
+  receiveAmount?: number | undefined;
+  error?: string;
+  warning?: string;
+  onChange: (value: string) => void;
+  onDebouncedChange: (value: string) => void;
+};
+
+/**
+ * Renders the input control to set the transaction amount
+ */
+const AmountInput = (props: Props) => {
+  const theme = useTheme();
+
+  const styles = useMemo(
+    () => ({
+      amountContainer: {
+        width: '100%',
+        maxWidth: '250px',
+      },
+      amountInput: {
+        background: theme.palette.input.background,
+        borderRadius: '8px',
+        border: 'none',
+      },
+      amountCardContent: {
+        display: 'flex',
+        alignItems: 'center',
+        height: '50px',
+        padding: 0,
+        ':last-child': {
+          padding: 0,
+        },
+      },
+      balance: {
+        color: theme.palette.text.secondary,
+        fontSize: '14px',
+        lineHeight: '14px',
+        textAlign: 'right',
+      },
+    }),
+    [theme],
+  );
+
+  const {
+    fromChain: sourceChain,
+    toChain: destChain,
+    isTransactionInProgress,
+  } = useSelector((state: RootState) => state.transferInput);
+
+  const { sourceToken, destToken } = useGetTokens();
+
+  const isInputDisabled = useMemo(
+    () =>
+      isTransactionInProgress ||
+      !sourceChain ||
+      !sourceToken ||
+      !destChain ||
+      !destToken,
+    [destChain, destToken, isTransactionInProgress, sourceChain, sourceToken],
+  );
+
+  return (
+    <Box sx={styles.amountContainer}>
+      <Card sx={styles.amountInput}>
+        <CardContent sx={styles.amountCardContent}>
+          <DebouncedTextField
+            fullWidth
+            disabled={isInputDisabled}
+            placeholder="0"
+            slotProps={{
+              htmlInput: {
+                style: {
+                  color: props.error
+                    ? theme.palette.error.main
+                    : theme.palette.text.primary,
+                  fontSize: '36px',
+                  height: '36px',
+                },
+                onWheel: (e) => {
+                  // IMPORTANT: We need to prevent the scroll behavior on number inputs.
+                  // Otherwise it'll increase/decrease the value when user scrolls on the input control.
+                  // See for details: https://github.com/mui/material-ui/issues/7960
+                  e.currentTarget.blur();
+                },
+                step: '0.1',
+              },
+              input: {
+                disableUnderline: true,
+              },
+            }}
+            variant="standard"
+            value={props.debauncedValue}
+            onChange={props.onChange}
+            onDebouncedChange={props.onDebouncedChange}
+          />
+        </CardContent>
+      </Card>
+    </Box>
+  );
+};
+
+export default React.memo(AmountInput);
```

Comment:
We are using React.xxx in some places and xxx directly, I think we should decide on what we want to do consistently